File: 1-two-sum/two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    const sumMap = new Map()
    for(let i = 0; i < nums.length; i++){
        let diff = target - nums[i]

        if(sumMap.has(diff)){
            return [i, sumMap.get(diff)]
        }

        sumMap.set(nums[i], i)
    }
    
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        sum_map = {}
        for i in range(len(nums)):
            diff = target - nums[i]

            if diff in sum_map:
                return[i, sum_map[diff]]
            
            sum_map[nums[i]] = i
        